plugins/clean.py

```diff
@@ -17,31 +17,47 @@ def handle(self, directive, data):
 
     def _process_clean(self, targets):
         success = True
-        for target in targets:
-            success &= self._clean(target)
+        defaults = self._context.defaults().get(self._directive, {})
+        force = defaults.get('force', False)
+        for target in self._dispatch_targets(targets):
```

Comment:
You can actually just replace this with `for target in targets` and that'll work fine for both dicts and lists (a for-in iteration over a dictionary iterates over its keys). And then you could get rid of the `_dispatch_targets` function.